collect_app/src/main/java/org/odk/collect/android/views/ODKView.java

```diff
@@ -483,7 +484,22 @@ public void highlightWidget(FormIndex formIndex) {
                 scrollTo(0, qw.getTop());
 
                 ValueAnimator va = new ValueAnimator();
-                va.setIntValues(getResources().getColor(R.color.red), getDrawingCacheBackgroundColor());
+                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT) {
+                    // only for kitkat and newer versions
+                    va.setIntValues(getResources().getColor(R.color.red),
+                            getDrawingCacheBackgroundColor());
+                    } else {
+                    TypedValue typedValue = new TypedValue();
+                    getContext().getTheme().resolveAttribute(android.R.attr.windowBackground, typedValue, true);
+                    if (typedValue.type >= TypedValue.TYPE_FIRST_COLOR_INT && typedValue.type <= TypedValue.TYPE_LAST_COLOR_INT) {
+                        // windowBackground is a color
```

Comment:
Comments are helpful when they explain WHY a certain piece of code exists. In this case, what is special about Android versions < 4.4 that make you want to do this check? Why is it important to know if the background is a color? In fact, what else can it be? Is there a reference that can help others understand why you have introduced this?